### DISCUSSION TITLE
Problem with training data format

### DISCUSSION THREAD

**Discussion by Tubakabura** [deepmodeling/dpgen#1423] [Q&A] [answered]
My initial training data was in 'mixed' format, while the data generated from each 'run' iteration was converted to the regular format of deepmd. 
Is there any way to control the format of the data generated in each iteration? Otherwise I had to run a python process to convert the data before each iteration.

**Comment by njzjz** [comment, marked as answer]
Alternately, you can set [`batch_size`](https://docs.deepmodeling.com/projects/deepmd/en/v2.2.7/train/train-input.html#argument:training/training_data/batch_size) to `mixed:N` (e.g. `mixed:16`) which will mix systems automatically during training.

If it can not meet your requirements, PR is welcome.

**Reply by chenggoj** [reply]
Hi.

I have the same problem. I am using the 0.13.1 version of DP-GEN. 

My initial dataset is mixed_type but when I finished the iter0 and update the dataset, it stopped the training process when starting iter1. 

The error is as followed:

> ==> /000/train.log <==
> , error_format_msg
> AssertionError: if one of the system is of mixed_type format, then all of the systems should be of mixed_type format!
>     assert data_sys.mixed_type, error_format_msg
> AssertionError: if one of the system is of mixed_type format, then all of the systems should be of mixed_type format!
>     assert data_sys.mixed_type, error_format_msg
> AssertionError: if one of the system is of mixed_type format, then all of the systems should be of mixed_type format!
>     assert data_sys.mixed_type, error_format_msg
> AssertionError: if one of the system is of mixed_type format, then all of the systems should be of mixed_type format!
> --------------------------------------------------------------------------
> prterun detected that one or more processes exited with non-zero status,
> thus causing the job to be terminated. The first process to do so was:
> 
>    Process name: [prterun-exp-8-60-2470089@1,2]
>    Exit code:    1
> --------------------------------------------------------------------------
> 

I have set the "batch_size" to `mixed:2`, but it did not work. I will have to convert the new data to mixed type mannually by myself and start the training again every time.
